model/src/DA/Finance/Asset.daml

```diff
@@ -47,6 +56,7 @@ template AssetDeposit
             -- needs to be smaller or equal than the current quantity. If it does not add up,
             -- an asset deposit with the remainder is created.
         do
+          assertNotLocked this
```

Comment:
I think this is too restrictive, as eg. the exchange will have to split a locked asset to process a partial fill. I think I have a remedy in my next commit where I'm trying to simplify the locking mechanism down to the very minimum